src/main/java/carpetextra/utils/RulePrinter.java

```diff
@@ -10,7 +10,7 @@
     @Override
     public void onInitialize() {
         System.setOut(CarpetRulePrinter.OLD_OUT);
-        CarpetExtraServer.noop();
+        CarpetServer.manageExtension(new CarpetExtraServer());
```

Comment:
You should call `onInitialize()` here. Even if it does exactly that, maybe at some point extra instructions are added, or the class name is changed, etc.

Reply:
To call `onInitialize()` would require doing `new CarpetExtraServer().onInitialize();`, which seems pointless since I can just have `new CarpetExtraServer()` here. Plus there might be things in onInitialize that don't need to be loaded to print the list of rules. I can move this line to a static method in CarpetExtraServer if that would work better though

Reply:
The new instance is basically what loader would do anyway, that's at least what I would have done. Believe me, it's better to run it as if it was a natural startup just before, else it can cause the action to fail in the future (this is for the rules-to-readme action that keeps the Readme updated, where Loader doesn't know about the other entrypoint since fabric.mod.json is completely replaced)